python/desc/twinkles/sprinkler.py

```diff
@@ -185,255 +184,317 @@ def __init__(self, catsim_cat, visit_mjd, specFileMap, sed_path,
+                row[self.defs_dict['galaxyBulge_is_sprinkled']] = 1
+                row[self.defs_dict['galaxyDisk_is_sprinkled']] = 1
+
+            #Replace original entry with new entry
+            input_catalog[rowNum] = row
+
+        for rowNum in valid_sne:
+            row = input_catalog[rowNum]
+            galtileid = row[galid_dex]
+            if self.cached_sprinkling is True:
+                if galtileid in self.sne_cache['galtileid'].values:
+                    use_system = self.sne_cache.query('galtileid == %i' % galtileid)['twinkles_system'].values
                     use_df = self.sne_catalog.query('twinkles_sysno == %i' % use_system)
+                    self.used_systems.append(use_system)
+                else:
+                    continue
+            else:
+                lens_sne_candidates = self.find_sne_lens_candidates(row[self.defs_dict['galaxyDisk_redshift']])
+                candidate_sysno = np.unique(lens_sne_candidates['twinkles_sysno'])
+                num_candidates = len(candidate_sysno)
+                if num_candidates == 0:
+                    continue
+                used_already = np.array([sys_num in self.used_systems for sys_num in candidate_sysno])
+                unused_sysno = candidate_sysno[~used_already]
+                if len(unused_sysno) == 0:
+                    continue
+                rng2 = np.random.RandomState(galtileid % (2^32 -1))
+                use_system = rng2.choice(unused_sysno)
+                use_df = self.sne_catalog.query('twinkles_sysno == %i' % use_system)
+
+            default_lensrow = row.copy()
+            default_lensrow[self.defs_dict['galaxyDisk_majorAxis']] = 0.0
+            default_lensrow[self.defs_dict['galaxyDisk_minorAxis']] = 0.0
+            default_lensrow[self.defs_dict['galaxyDisk_positionAngle']] = 0.0
+            default_lensrow[self.defs_dict['galaxyDisk_internalAv']] = 0.0
+            default_lensrow[self.defs_dict['galaxyDisk_magNorm']] = 999. #np.nan To be fixed post run1.1
+            default_lensrow[self.defs_dict['galaxyDisk_sedFilename']] = None
+            default_lensrow[self.defs_dict['galaxyBulge_majorAxis']] = 0.0
+            default_lensrow[self.defs_dict['galaxyBulge_minorAxis']] = 0.0
+            default_lensrow[self.defs_dict['galaxyBulge_positionAngle']] = 0.0
+            default_lensrow[self.defs_dict['galaxyBulge_internalAv']] = 0.0
+            default_lensrow[self.defs_dict['galaxyBulge_magNorm']] = 999. #np.nan To be fixed post run1.1
+            default_lensrow[self.defs_dict['galaxyBulge_sedFilename']] = None
+            varString = 'None'
+            default_lensrow[self.defs_dict['galaxyAgn_varParamStr']] = varString
+
+            for i in range(len(use_df)):
+                lensrow = default_lensrow.copy()
+                delta_ra = np.radians(use_df['x'].iloc[i] / 3600.0)
+                delta_dec = np.radians(use_df['y'].iloc[i] / 3600.0)
+                if use_system == 1953:
```

Comment:
Is this meant to print in production? It was probably for testing/debugging I'm guessing.

Reply:
Good catch.  I will remove that